### PR TITLE
Clarify how to use kubeconfig

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/clusterctl/README.md [hunk 1]
```diff
@@ -66,9 +66,9 @@ Once you have created a cluster, you can interact with the cluster and machine
 resources using kubectl:
 
 ```
-$ kubectl --kubeconfig kubeconfig get clusters
-$ kubectl --kubeconfig kubeconfig get machines
-$ kubectl --kubeconfig kubeconfig get machines -o yaml
+$ kubectl get clusters
+$ kubectl get machines
+$ kubectl get machines -o yaml
```
Comment (jichenjc) [inline cmd/clusterctl/README.md]: I thought by default we will use ~/.kube/config ? or it will fall back to kubeconfig on $PWD?

**Comment by gyliu513** [comment]
@vincepri @jichenjc The reason that I want to propose this PR is because with KIND, I already exposed `KUBECONFIG`, so there is no need to specify `--kubeconfig` in `kubectl`, here I just updated the PR by adding a **NOTE** to clarify this. Hope this helps, thanks!

File: cmd/clusterctl/README.md [hunk 2]
```diff
@@ -71,6 +71,8 @@ $ kubectl --kubeconfig kubeconfig get machines
 $ kubectl --kubeconfig kubeconfig get machines -o yaml
 ```
 
+**NOTE:** There is no need to specify `--kubeconfig` if your `kubeconfig` was located in default directory under `$HOME/.kube/config` or if you already exposed env variable `KUBECONFIG`.
```
Comment (bradtopol) [inline cmd/clusterctl/README.md]: Some minor tweaks:  
"was located in default directory" should be replaced with:
"was located in the default directory"

"you already exposed env variable" should be replaced with
"you have already exposed the env variable"

**Comment by bradtopol** [comment]
I made some minor tweaks to improve readability. See inline comment

File: cmd/clusterctl/README.md (hunk 2 above)
Comment (jichenjc) [reply cmd/clusterctl/README.md]: +1 for the wording

Comment (gyliu513) [reply cmd/clusterctl/README.md]: Done, thanks @bradtopol and @jichenjc

**Pull request merged**